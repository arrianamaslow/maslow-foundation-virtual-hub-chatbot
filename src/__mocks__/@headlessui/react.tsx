/**
 * @jest-environment jsdom
 */

import { DialogProps } from '@headlessui/react'
import React, { ReactNode } from 'react'

type DialogPanelTestProps = {
  children?: ReactNode
}

interface DialogTestProps extends DialogProps {
  children?: ReactNode
}

const DialogPanel: React.FC<DialogPanelTestProps> = ({ children }) => {
  return <div data-testid='mock-dialog-panel-component'>{children}</div>
}

const Dialog: React.FC<DialogTestProps> = ({ open, children }) => {
  return (
    <div data-testid='mock-dialog-component'>
      {open ? 'DIALOGUEOPENED' : 'DIALOGUECLOSED'}
      {children}
    </div>
  )
}

export { Dialog, DialogPanel }
