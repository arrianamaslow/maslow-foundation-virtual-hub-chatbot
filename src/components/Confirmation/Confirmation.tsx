import Image from 'next/image'

type ConfirmationProps = {
  date: string
  time: string
}

const Confirmation: React.FC<ConfirmationProps> = ({ date, time }) => {
  return (
    <div className='flex text-center text-[1.75rem] flex-col items-center '>
      <p className='gap-3 p-3'>
        {date && time ? (
          <>
            Booking confirmed! See you on {date}, at {time} (UK time)
          </>
        ) : (
          <>Booking confirmed!</>
        )}
      </p>
      <div>
        <Image src='/img/confirmation-tick.svg' alt='my image' width={190} height={190} />
      </div>
    </div>
  )
}

export default Confirmation
