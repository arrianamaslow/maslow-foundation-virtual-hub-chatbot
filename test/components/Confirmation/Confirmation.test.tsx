/**
 * @jest-environment jsdom
 */
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import Confirmation from '@/components/Confirmation/Confirmation'

describe('<Confirmation />', () => {
  it('renders correctly', () => {
    const tree = render(<Confirmation date='August 2, 216 BCE' time='13:48' />)
    expect(tree).toMatchSnapshot()
  })
})
