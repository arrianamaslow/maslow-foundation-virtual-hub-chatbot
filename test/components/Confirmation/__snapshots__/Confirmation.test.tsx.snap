// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Confirmation /> renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex text-center text-[1.75rem] flex-col items-center "
      >
        <p
          class="gap-3 p-3"
        >
          Booking confirmed! See you on 
          August 2, 216 BCE
          , at 
          13:48
           (UK time)
        </p>
        <div>
          <img
            alt="my image"
            height="190"
            src="/img/confirmation-tick.svg"
            width="190"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex text-center text-[1.75rem] flex-col items-center "
    >
      <p
        class="gap-3 p-3"
      >
        Booking confirmed! See you on 
        August 2, 216 BCE
        , at 
        13:48
         (UK time)
      </p>
      <div>
        <img
          alt="my image"
          height="190"
          src="/img/confirmation-tick.svg"
          width="190"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
